vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/variable/ClipboardTextDynamicVariable.java

```diff
@@ -16,26 +16,77 @@
 
 package org.vividus.bdd.mobileapp.variable;
 
-import javax.inject.Named;
+import static org.apache.commons.lang3.Validate.isTrue;
+
+import com.google.common.eventbus.Subscribe;
 
 import org.vividus.bdd.variable.DynamicVariable;
+import org.vividus.mobileapp.action.ApplicationActions;
 import org.vividus.selenium.IWebDriverProvider;
+import org.vividus.selenium.event.AfterWebDriverQuitEvent;
+import org.vividus.selenium.manager.IGenericWebDriverManager;
+import org.vividus.testcontext.TestContext;
 
+import io.appium.java_client.HasSessionDetails;
 import io.appium.java_client.clipboard.HasClipboard;
 
-@Named("clipboard-text")
 public class ClipboardTextDynamicVariable implements DynamicVariable
 {
+    private static final Object KEY = BundleIdKey.class;
+
     private final IWebDriverProvider webDriverProvider;
+    private final IGenericWebDriverManager genericWebDriverManager;
+    private final TestContext testContext;
+    private final ApplicationActions applicationActions;
+
+    private final boolean realDevice;
+    private final String webDriverAgentBundleId;
 
-    public ClipboardTextDynamicVariable(IWebDriverProvider webDriverProvider)
+    public ClipboardTextDynamicVariable(boolean realDevice, String webDriverAgentBundleId,
+            IWebDriverProvider webDriverProvider, IGenericWebDriverManager genericWebDriverManager,
+            TestContext testContext, ApplicationActions applicationActions)
     {
         this.webDriverProvider = webDriverProvider;
+        this.genericWebDriverManager = genericWebDriverManager;
+        this.testContext = testContext;
+        this.applicationActions = applicationActions;
+        this.realDevice = realDevice;
+        this.webDriverAgentBundleId = webDriverAgentBundleId;
     }
 
     @Override
     public String getValue()
+    {
+        if (realDevice && genericWebDriverManager.isIOSNativeApp())
```

Comment:
is the logic below applicable only for native tests and not for web ones?

Reply:
- Not sure about iOS, but on android appium will spin up chromium driver, need to check what driver is used for iOS
- In case of web tests through appium I assume that end-users run tests in this way to check adaptivity or just cross platform testing, so initial tests were written in selenium where we do not an easy way to get the clipboard text